68370622 upgrade to bootstrap 3.x

bryszard opened the pull request:
Refers to PT - https://www.pivotaltracker.com/story/show/68370622

I've changed / removed all obsolete classes. However, the display needs now an adjustment:

![image](https://cloud.githubusercontent.com/assets/12682792/21059540/1f8a981a-be43-11e6-936a-84972438be58.png)

tansaku commented:
thanks for this @bryszard - great start!

I see we've got a few cucumber tests failing and the styling issues you mention - I guess this might become epic ...

tansaku commented:
@bryszard did you want to create an epic for this so we can start listing up all the separate issues that need to be addressed in the updated interface ...?

bryszard commented:
@tansaku No, I wanted to do it myself. You think an epic is needed?

I tried to update the classes / remove the obsolete classes, but the
display is not good enough. So my current idea is to extract the old css
classes to our css file and keep it as it is. What do you think about it?

2016-12-17 15:19 GMT+01:00 Sam Joseph <notifications@github.com>:

> @bryszard <https://github.com/bryszard> did you want to create an epic
> for this so we can start listing up all the separate issues that need to be
> addressed in the updated interface ...?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/AgileVentures/LocalSupport/pull/408#issuecomment-267765218>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AMGGKAbW-G8gGVsbyqGk1lkAM6Vg4D8dks5rI-9cgaJpZM4LJNWw>
> .
>

tansaku commented:
I was just thinking epic to list out all the different visual fixes that were needed so they could be worked on by different people in parallel ...

extraction of css classes sounds like a good idea.  We don't have to keep the exact same look and feel, but we do need to make it look okay.

Maybe this is something we need to look at together in more detail ...

bryszard commented:
@tansaku I'm trying to compare the old page and adjust the new page, copying the old classes and css attributes. It's quite a work - I was working today 2 hours just on header and still didn't finish the login popup ;). Copying old css classes produces also some new CC complaints, so this will need to be fixed too.

Anyway I will slowly push it through. I don't have energy to create an epic and split the work for now. However, if you want you can do it.

bryszard commented:
@tansaku @Maroo-b  Ok, I went through all the app and adjusted everything I've found. Front page looks now like that (the map size is not my fault ;). I've found that it's result of another PR - https://github.com/AgileVentures/LocalSupport/pull/397):

![image](https://cloud.githubusercontent.com/assets/12682792/21753830/21145b48-d5f5-11e6-9e1b-cbdf887679b1.png)

tansaku commented:
great work @bryszard !  

map size doesn't look too bad to me :-) 

So is this ready for a full review?

tansaku commented:
I guess it must be since there is a request for review :-) I've got to get into this new flow ... :-)

johnnymo87 commented on spec/javascripts/this_is_my_org_spec.js:
```diff
@@ -21,6 +21,11 @@ describe('This is my Organisation button', function() {
         register = $('#registerForm');
+        timelapse = function() {
+          setTimeout(function() {
+            x = 5;
```
What does this do?

bryszard replied:
Nothing, shouldn't be there ;). I was experimenting with stopping time in Jasmine specs. Deleting it.

bryszard commented on spec/javascripts/this_is_my_org_spec.js:
```diff
@@ -41,11 +46,8 @@ describe('This is my Organisation button', function() {
-            });
             it('nav changes attributes', function() {
-                expect(nav).toHaveClass('in');
+                expect(nav).toHaveClass('collapsing');
```
By the way I had here a problem with Jasmine spec and didn't know how to fix it.

Manually js works just fine, but on Jasmine the element `.nav-collapse` is stuck in between state ('collapsing' is added when the collapsing starts, but should be changed to 'collapse in' when collapsing finishes). Also, for some reason the 'show' even is not shown as Triggered on 'nav' although before it is tested that 'show' is being called.

tansaku replied:
interesting - but all the jasmine tests are still passing in CI ...

bryszard commented:
@tansaku Yes, it's ready for full review :). I don't know if I didn't miss some views, but I tried to cover it all.

tansaku commented:
Great work @bryszard - really appreciate you working so hard on this.

I've got it running locally and there are a few things I notice.  Don't feel that you've got to fix each and every one - maybe it's something we turn into separate tickets as part of the bigger epic, but if you can see simple ways to fix, do go ahead:

1) I notice the login box organisation has changed - looks a bit ragged

![login box layout](https://www.dropbox.com/s/xin1r91plc8ngpi/Screenshot%202017-01-12%2010.53.31.png?dl=1)

2) in the map the larger icons are appearing behind the small ones:

![larger icons hidden](https://www.dropbox.com/s/iokqslrt1q6advm/Screenshot%202017-01-12%2010.55.28.png?dl=1)

3) you mentioned already the map is not as high - would be great to fix that

4) I think I see a bug that volops individual show pages are not focusing on the individual volop - but that's in the production, so don't think it's related - just noticed it here ...

I think that's everything I saw in my review ... I notice the font/text looks much sharper and cleaner - really want to get this deployed - I guess there's also checking how it looks on mobile ... although we never optimised the site for that ... :-(

tansaku commented:
quick review on mobile suggests we are no worse than we ever were - that's another epic right there - produce a mobile friendly version ...

bryszard commented:
@tansaku Thanks for your comments, I'll look at it during the weekend. About the 3. - ok, I'll make it back to the point as it is on production ;).

tansaku commented:
thanks @bryszard - I note that maybe some of these issues are not specific to your PR - I'm seeing that dot issue with overlap in the map in my local develop branch ...

tansaku commented:
@Maroo-b I just ran git bisect and it seems like the change in the ordering of the markers occurred when we dropped that gem:

```
→ git bisect good 8420db1
Bisecting: 2 revisions left to test after this (roughly 2 steps)
[ab5a6d23e7f9852d47d966da363d2c27cf5756eb] 132104697 method to user model (#407)
[tansaku@Samuels-MBP:~/Documents/Github/AgileVentures/LocalSupport ((ab5a6d2...)|BISECTING)]$ 
→ git bisect bad
Bisecting: 0 revisions left to test after this (roughly 1 step)
[9d534c582c8d8a6cbb82fbc2990dc285969b4069] Remove gmaps4rails gem [fixes #134262481] (#397)
[tansaku@Samuels-MBP:~/Documents/Github/AgileVentures/LocalSupport ((9d534c5...)|BISECTING)]$ 
→ git bisect bad
Bisecting: 0 revisions left to test after this (roughly 0 steps)
[48f189cdf2bb49377e8068e09ce0611a82ade822] Improve formatting in c9.md (#414)
[tansaku@Samuels-MBP:~/Documents/Github/AgileVentures/LocalSupport ((48f189c...)|BISECTING)]$ 
→ git bisect good
9d534c582c8d8a6cbb82fbc2990dc285969b4069 is the first bad commit
commit 9d534c582c8d8a6cbb82fbc2990dc285969b4069
Author: Marouen Bousnina <marouen.it.technology@gmail.com>
Date:   Fri Dec 16 13:10:03 2016 +0100

    Remove gmaps4rails gem [fixes #134262481] (#397)
    
    * remove gmaps4rails gem [fixes #134262481]
    
    Addresses: https://www.pivotaltracker.com/story/show/134262481
    
    * Add @maps tags and fix code climate warning

:100644 100644 e53b3b5c3a8c174701766e434081a07e9aa9f4f6 97794f72d9055527cbf246295bcef2124c72213b M	Gemfile
:100644 100644 0a73ec91e2a547a8eed06eff6875e2f9e18b6920 7002980c476564fe59b67b1a17ea162384c66a8b M	Gemfile.lock
:040000 040000 4649ae052281ffcd46d77f090e70820dfd285836 8d407c7a0ea3cd55c1986d5c3940f235c60dc622 M	app
:100644 100644 c572bba5970869a78f34b8bc9320f56f1a766f18 dc8bf08c81de5038e5f401e58425aeb2315d29c1 M	bower.json
:040000 040000 1c4d966c42245073a41f591e3d08ccda65b4278d 41582e71ef10f05b738f4e9fa31f235b1841d8f1 M	features
:040000 040000 311ce7e5ca99e66d850a87f942a085fa0bf9bb2b 2c97b0a29c11f5ed33195f12183b41ececd38ae7 M	spec

```

I missed that in the review ...

bryszard commented:
@tansaku 

1. I've checked it locally and I cannot reproduce the problem, my login box looks ok:

![image](https://cloud.githubusercontent.com/assets/12682792/21955780/2aaca1de-da72-11e6-8107-78f5a11ad6a9.png)

2. As you noticed, it was removing the gem which caused this.

3. The size of the map is also caused by the removal of gmaps4rails gem. I'll try to change the size anyway.

4. As I understand I don't need to do anything about it for now?

bryszard commented:
@tansaku 

2. I see that @Maroo-b fixed this issue here - #422 

3. I've fixed it. There was no height regulation in new js. I've added it:

![image](https://cloud.githubusercontent.com/assets/12682792/21956149/6721c4c0-da7a-11e6-98a7-29b8ddb41837.png)

tansaku commented:
thanks @bryszard - I think you've addressed everything, and that login layout thing isn't occurring for me on this updated branch ... ah - I think it occurs for me at 75% zoom in chrome:

![](!https://www.dropbox.com/s/bxy3dgukoftxkq2/Screenshot%202017-01-16%2015.36.53.png?dl=1)

but not in production

![](https://www.dropbox.com/s/rys44ut8g6ssfak/Screenshot%202017-01-16%2015.35.50.png?dl=1)

I notice we've lost the curved corners on those input fields as well - would that be an easy fix?

Anyway, try zooming in and out on the login box on your system to see if you can replicate (I'm in chrome) 

I guess ideally we'd be testing across a range of browsers, but that should be another ticket to get set up with saucelabs - now we're an official charity, perhaps we can get set up with some free cross-browser testing action ...?

bryszard commented:
@tansaku I still cannot reproduce this position problem - I checked in all zooms and also in Mozilla.

But I've adjusted the border-radius and also position of the checkbox:

Chrome, zoom 75%:
![image](https://cloud.githubusercontent.com/assets/12682792/22036655/8c5fb0a4-dcf4-11e6-9aab-989ee46580eb.png)

Chrome, zoom 125%:
![image](https://cloud.githubusercontent.com/assets/12682792/22036600/5b29af6c-dcf4-11e6-9332-6e146d4dba48.png)

Chrome, zoom 175%:
![image](https://cloud.githubusercontent.com/assets/12682792/22036622/6d14768a-dcf4-11e6-925e-9c41a13e1350.png)

tansaku commented:
@bryszard weird - well great work adjusting the radius etc. if it's just a corner case on my local machine I guess we get this checked in and move on - looking forward to our CC score going up
